sdk/core/core-lro/src/lroEngine/operation.ts

```diff
@@ -122,6 +123,7 @@ export class GenericPollOperation<TResult, TState extends PollOperationState<TRe
 
   async cancel(): Promise<PollOperation<TState, TResult>> {
     this.state.isCancelled = true;
+    await this.cancelOp?.(this.state);
```

Comment:
Calling polling will throw an error with the message `Poller cancelled`. There is one such test case in the TA PR. This is coming from the abstract `Poller` class. I agree though we should add behavioral tests after cancellation, I opened https://github.com/Azure/azure-sdk-for-js/issues/21865 to track this for now since I think it is out of the scope of this PR.